app/models/user_batch.rb

```diff
@@ -35,79 +36,247 @@ def create_users(mission)
+            add_error(error, attribute, row_number)
           end
-          @validation_error = true
         end
+      end
+      @validation_error = true unless error_is_on_persistence_token? user
+    end
+  end
 
-        users << user
+  def error_is_on_persistence_token?(user)
+    (user.errors.keys.length == 1) && (user.errors.keys.include? :persistence_token)
+  end
 
-      end # iteration over rows
+  def add_error(error, attribute, row_number)
+    row_error = I18n.t('import.row_error', row: row_number, error: error)
+    errors.add("users[#{row_number}].#{attribute}", row_error)
+  end
 
-      # now if there was a validation error with any user, rollback the transaction
-      raise ActiveRecord::Rollback if @validation_error
+  def add_too_many_errors(row_number)
+    # we pass the previous row number to the error message formatting
+    # since the error count cutoff was surpassed by the previous rows,
+    # not this row
+    errors.add(:users, :too_many_errors, row: row_number - 1) if errors_reached_limit
+  end
 
-    end # transaction
+  def errors_reached_limit
+    errors.count >= IMPORT_ERROR_CUTOFF
+  end
 
-    return succeeded?
+  def actual_row_number(row_start, index)
+    index + row_start + 1
+  end
+
+  def check_uniqueness_on_objects(objects, field)
+    key = correct_field_key(field)
+    # Add errors on fields that aren't unique
+    @fields_hash_table[key].each do |k,v|
+      v.each { |i| objects.at(i).errors.add(field, :taken) } if v.length != 1
+    end
+  end
+
+  def check_uniqueness_on_db(objects, table, field, row_start, columns=[])
+    results = DirectDBConn.check_uniqueness_on_db(objects, table, field, row_start, columns)
+
+    results.each do |result|
+      @fields_hash_table[field][result.first].each do |index|
+        p "UNIQ FAIL DB: #{field} : #{result.first}"
+        object = objects.at(index)
+        object.errors.add(field, :taken)
+      end
+    end
+  end
+
+  def correct_field_key(field)
+    field.include?('phone') ? 'phone' : field
+  end
+
+  def insert_assignments_on_db(objects)
+    assignments = objects.map{ |u| u.assignments.first }
+
+    string_params = DirectDBConn.generate_string_params_template(assignments.length)
+    column_names = assignments.first.attributes.keys.join(', ')
+    users_id = objects.map{|o| "(SELECT id FROM users WHERE import_num=#{o.import_num})" }
+
+    object_values = assignments.map.with_index do |assignment, index|
+      values_in_string = DirectDBConn.convert_values_to_insert_syntax(assignment)
+      values_as_array = values_in_string.split(',')
+
+      #Add Select query on user_id position
+      user_id_index = column_names.split(', ').index('user_id')
+      values_as_array[user_id_index] = users_id[index]
+
+      "(#{values_as_array.join(',')})"
+    end
+
+    sql_query_array = ["INSERT INTO assignments (#{column_names}) VALUES #{object_values.join(', ')}"]
+
+    DirectDBConn.execute_sql_query_array(sql_query_array)
```

Comment:
I remember i had an issue doing a batch insert like that. For one query, the `INSERT ... SELECT` was working, but for more, no. I didn't want to loose more time on that, so i just switched to the other syntax that should be the same thing.

What I'm doing is the same thing, no? It's doing the `"(SELECT id FROM users WHERE import_num=#{o.import_num})"` all in one query, as it would be with `INSERT ... SELECT`. I think it's just a syntactic sugar for the same thing, no?